frontend/www/js/omegaup/arena/arena.ts

```diff
@@ -543,6 +538,93 @@ export class Arena {
       });
     }
 
+    // Setup run submit view, if it is available.
+    if (document.getElementById('run-submit') !== null) {
+      this.runSubmitView = new Vue({
+        el: '#run-submit',
+        render: function(createElement) {
+          return createElement('omegaup-arena-runsubmit', {
+            props: {
+              languages: this.languages,
+              submissionGapSecondsRemaining: this.submissionGapSecondsRemaining,
+              preferredLanguage: this.preferredLanguage,
+            },
+            on: {
+              'submit-run': (code: string, language: string) => {
+                this.submitRun(code, language);
```

Comment:
se puede mejor usar

```suggestion
                self.submitRun(code, language);
```

? (puedes declarar `const self = this;` arriba de L543). así `submitRun` se puede dejar en `class Arena`.

Reply:
Ohh, eso no se me ocurrió antes para no mover esa función. Gracias.